client/extensions/woocommerce/app/order/order-payment/dialog.js

```diff
@@ -0,0 +1,226 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	arePaymentMethodsLoaded,
+	getPaymentMethod,
+} from 'woocommerce/state/sites/payment-methods/selectors';
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import { fetchPaymentMethods } from 'woocommerce/state/sites/payment-methods/actions';
+import formatCurrency from 'lib/format-currency';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextarea from 'components/forms/form-textarea';
+import { getOrderRefundTotal } from 'woocommerce/lib/order-values';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import Notice from 'components/notice';
+import OrderRefundTable from './table';
+import PriceInput from 'woocommerce/components/price-input';
+import { sendRefund } from 'woocommerce/state/sites/orders/refunds/actions';
+
+class OrderPaymentCard extends Component {
+	static propTypes = {
+		isPaymentLoading: PropTypes.bool,
+		fetchPaymentMethods: PropTypes.func.isRequired,
+		isVisible: PropTypes.bool.isRequired,
+		order: PropTypes.shape( {
+			currency: PropTypes.string.isRequired,
+			id: PropTypes.number.isRequired,
+			refunds: PropTypes.array.isRequired,
+			status: PropTypes.string.isRequired,
+			total: PropTypes.string.isRequired,
+		} ),
+		paymentMethod: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ),
+		sendRefund: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+		toggleDialog: PropTypes.func,
+		translate: PropTypes.func.isRequired,
+	};
+
+	state = {
+		errorMessage: false,
+		refundTotal: 0,
+		refundNote: '',
+	};
+
+	componentDidMount = () => {
+		const { siteId } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchPaymentMethods( siteId );
+		}
+	};
+
+	componentWillReceiveProps = newProps => {
+		const newSiteId = newProps.siteId;
+		const oldSiteId = this.props.siteId;
+
+		if ( oldSiteId !== newSiteId ) {
+			this.props.fetchPaymentMethods( newSiteId );
+		}
+	};
+
+	toggleDialog = () => {
+		this.setState( {
+			errorMessage: false,
+			refundTotal: 0,
+			refundNote: '',
+		} );
+		this.props.toggleDialog();
+	};
+
+	recalculateRefund = total => {
+		this.setState( { refundTotal: total } );
+	};
+
+	updateNote = event => {
+		this.setState( {
+			refundNote: event.target.value,
+		} );
+	};
+
+	sendRefund = () => {
+		const { order, paymentMethod, siteId, translate } = this.props;
+		const maxRefund = parseFloat( order.total ) + getOrderRefundTotal( order );
```

Comment:
Maybe add a comment here that getOrderRefundTotal is always negative. Otherwise this looks a little funny.